build/index.js

```diff
@@ -65,6 +66,18 @@ async function main() {
 
     const eslintOptions = ["--ext", ".ts,.js,.tsx,.jsx", "."];
 
+    const rootInstallDone = existsSync("./node_modules");
+    if (
+        !rootInstallDone &&
+        ["lint", "lintfix", "clean", "install", "installAndBuild"].includes(
```

Comment:
Hm, no, not really. Since some stuff that's needed to install is actually at the root, they both need the root install to be there or it will install all over dependencies that should be in the root.

Reply:
Can eliminate the constant, though...

Reply:
My point is: If the command is `install`,  I'd expect it to run `npm ci` on the root modules no matter what - for example after pulling changes or switching branches. Without this, the only way to ensure the modules are reinstalled is to run `clean` - or to remove the root _node_modules_ manually. Given how `install` takes quite a bit of time anyway, I think forcefully reinstalling root modules while at it is worth it.

Reply:
Well, I'd go for that if root install didn't take a good amount of seconds (which is ever increasing). But that's a  rule of thumb that need not hold. We'd need to check if package-lock.json is newer than node_modules. So it's either a (possibly faulty) assumption, or add more state checking... I don't know. In any case, it's added complexity. I'd leave this running for a time, and if we find a pressing need, let's raise the issue and change it.

Reply:
Ah, OK, I see what you mean with changing branches; in that case node_modules can be out of sync, but detected so it will not run. Still, it's such an edge case that I'd rather go for devs deleting node_modules and running build than to add the penalty of running npm ci and adding 6-8 seconds every time you run build _and_ buildAndInstall.

Reply:
Well, `build.js build` shouldn't be affected, only `build.js` without arguments or with the `installAndBuild` command. Given how that takes >2 minutes on my machine, I think adding 6-8 seconds to avoid difficult to track down bugs seems fair...